src/hotspot/share/runtime/stackOverflow.cpp

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "logging/log.hpp"
+#include "runtime/os.inline.hpp"
+#include "runtime/stackOverflow.hpp"
+#include "runtime/thread.hpp"
+#include "utilities/align.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+size_t StackOverflow::_stack_red_zone_size = 0;
+size_t StackOverflow::_stack_yellow_zone_size = 0;
+size_t StackOverflow::_stack_reserved_zone_size = 0;
+size_t StackOverflow::_stack_shadow_zone_size = 0;
+
+void StackOverflow::initialize_stack_zone_sizes(size_t alignment) {
```

Comment:
I think you can remove the parameter and hard-code 4K inside this function. The reason is that the "StackXXXPages" parameters are defined as "number of 4k units", not "number of pages".

Reply:
Ok, I'll move the comment in os.cpp also that tells you why 4k is a thing.

Reply:
Ok, I made the suggested change and updated the PR.